### ISSUE TITLE
Open a new file from terminal from same project, clears old tabs and unsaved buffer from that project

### ISSUE THREAD

**Issue by ungb** [atom/atom#13729]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * [x] Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * [x] Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * [x] Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * [x] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * [x] Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
Open a new file from terminal from same project, clears old tabs and unsaved buffer from that project

### Steps to Reproduce
1. Open a project in terminal with `atom .`
2. Open a file and modify it, don't save.
3. Close Atom(this won't repro if there's other atom windows open.  Only if all Atom instances are closed)
4. On the same directory in terminal, do `atom <new file name>`

**Actual**: Atom opens new file with the current folder.  The changes on the file that was modify in step 2 are gone.  When you close Atom and reopen with `atom .` or opening recent projects, it shows the new file from step 4 only.

**Expected**: old tabs and unsaved changes should exist(if there are any) and new file should open in new tab 

**Reproduces how often:** 100% if you don't have another atom instance open, If you do, the new file will be opened there.

### Versions

```
atom --version
Atom    : 1.13.1
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

```
apm --version
apm  1.14.1
npm  3.10.5
node 4.4.5
python 2.7.10
git 2.10.1
```

This issue was found while testing https://github.com/atom/tabs/issues/306#issuecomment-276861137

**Comment by iolsen** [comment]
I thought this was in the category of problems Reopen Project is supposed to fix but I can reproduce this. The new file is opened in the scope of the project and the pending changes are lost.

Good find, @ungb.

**Comment by nathansobo** [comment]
This was spuriously closed via an incorrect "closes" comment in commit c6cae5b.
